### DISCUSSION TITLE
Concatenate IVT from MERRA2 reanalysis

### DISCUSSION THREAD

**Discussion by laisgfernandes** [SciTools/iris#5072] [Q&A] [answered]
Hello all,

I am trying to concatenate MERRA2 files using iris, but I am having problems. Here is my code:

    #####################################
    """Concatenate IVT from the MERRA2
    reanalysis."""

    import iris
    import numpy as np
    import matplotlib as mpl
    from iris.util import unify_time_units
    from iris.util import equalise_attributes

    def clean_callback(cube,field,filename):
        cube.coord('time').attributes = {}

    #read meridional IVT file.
    IVT_file = '/Volumes/climate_lab2/MERRA2/Daily_and_Subdaily/IVT_daily/MERRA2.tavg1_2d_int_Nx.20180101.SUB.nc'

    IVT = iris.load_cube(IVT_file, 'northward_flux_of_atmospheric_water_vapor', callback=clean_callback)

    dd = np.arange('2018-01-02', '2019-01-01', dtype='datetime64')

    dd = np.datetime_as_string(dd)

    hyphenlist = dd
    newlist = []

    for x in hyphenlist:
        newlist.append(x.replace('-', ''))

    dd = newlist

    for i in dd:
        print(i)
        #read meridional IVT file.
        IVT_file = '/Volumes/climate_lab2/MERRA2/Daily_and_Subdaily/IVT_daily/MERRA2.tavg1_2d_int_Nx.'+str(i)+'.SUB.nc'
        IVT_n = iris.load_cube(IVT_file, 'northward_flux_of_atmospheric_water_vapor', callback=clean_callback)

        cube = iris.cube.CubeList([IVT,IVT_n])
        iris.util.unify_time_units(cube)
        equalise_attributes(cube)
        IVT = cube.concatenate_cube()

    iris.save(IVT,'/Users/lais3/Desktop/IVT_conc/IVT_merid/MERRA2.tavg1_2d_int_Nx.2018.nc')
    #######################################

Python is returning:
ConcatenateError: failed to concatenate into a single cube.
Dimension coordinates metadata differ: time != time

I believe the issue is about the dtype of the time coordinate. After type:

    print(IVT.coord('time'))

Python shows me that the dtype for time is int32. For older files, it was float64, and I was being able to concatenate.

Please, I need to learn how I change the dtype from int32 to float64. I couldn't find an explanation in the iris user guide.

Many thanks for your assistance.
Cheers,
Laís

**Comment by trexfeathers** [comment, marked as answer]
Hi @laisgfernandes. The values in a `DimCoord` ([`DimCoord.points`](https://scitools-iris.readthedocs.io/en/stable/generated/api/iris/coords.html#iris.coords.DimCoord.points)) are a standard NumPy `ndarray`, so the correct way of changing type is to use [`ndarray.astype()`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html) on the `DimCoord`'s `points`, e.g.:

```python
import numpy as np

my_coord = IVT.coord("time")
my_coord.points = my_coord.points.astype(np.float64)
```

**Reply by laisgfernandes** [reply]
Hi @trexfeathers,

Thanks for your reply! I know I had to use astype for NumPy. I forgot to use .points to apply Numpy functions. Thank you very much for remembering me. I could concatenate the data after that.

Cheers,
Laís
